freqtrade/rpc/telegram.py

```diff
@@ -129,7 +129,7 @@ def _init_keyboard(self) -> None:
             r'/weekly$', r'/weekly \d+$', r'/monthly$', r'/monthly \d+$',
             r'/forcebuy$', r'/forcelong$', r'/forceshort$',
             r'/forcesell$', r'/forceexit$',
-            r'/edge$', r'/health$', r'/help$', r'/version$'
+            r'/edge$', r'/health$', r'/help$', r'/version$', r'/marketdir \d+$'
```

Comment:
`\d+` is wrong - as that is regex speak for a number - but we expect `"long", "short", "even", ...`).

Possible solutions will be either generally allowing a string `/marketdir \s+$` - or limiting to allowed strings  `/marketdir (long|short|...)$` (both solutions untested - but most likely somewhat working).

I'd suggest to actually try this (this defines the "valid" entries for [telegram keyboard config](https://www.freqtrade.io/en/stable/telegram-usage/#create-a-custom-keyboard-command-shortcut-buttons)).
You can also use [regex101](https://regex101.com/) to make finding the correct regex easier.

Reply:
Thanks for the pointers!

I was just a bit confused by what you meant in the last part. Do you suggest implementing the regex you wrote in place of what I have? I didn't quite get what you meant by 

> (this defines the "valid" entries for [telegram keyboard config](https://www.freqtrade.io/en/stable/telegram-usage/#create-a-custom-keyboard-command-shortcut-buttons))

Could you elaborate a bit on what you meant by this?

Reply:
"this" -> the regex used.

there's a pretty well defined list of acceptable commands - (e.g. `/marketdir long`) - which are currently disallowed (\d+ matches numbers only).
`/marketdir 2` (which the current would allow) is not among the actually working commands ...